cpp/arduino/Wire.h

```diff
@@ -1,138 +1,265 @@
+/*
+ * The Wire Library (https://www.arduino.cc/en/Reference/Wire)
+ * allows you to communicate with I2C/TWI devices. The general
+ * TWI protocol supports one "master" device and many "slave"
+ * devices that share the same two wires (SDA and SCL for data
+ * and clock respectively).
+ *
+ * You initialize the library by calling begin() as a master or
+ * begin(myAddress) as a slave (with an int from 8-127). In the
+ * initial mock implementation we support only the master role.
+ *
+ * To send bytes from a master to a slave, start with
+ * beginTransmission(slaveAddress), then use write(byte) to
+ * enqueue data, and finish with endTransmission().
+ *
+ * When a master wants to read, it starts with a call to
+ * requestFrom(slaveAddress, quantity) which blocks until the
+ * request finishes. The return value is either 0 (if the slave
+ * does not respond) or the number of bytes requested (which
+ * might be more than the number sent since reading is simply
+ * looking at a pin value at each clock tick).
+ *
+ * A master can write to or read from two or more slaves in
+ * quick succession (say, during one loop() function), so our
+ * mock needs to support preloading data to be read from multiple
+ * slaves and archive data sent to multiple slaves.
+ *
+ * In the mock, this is handled by having an array of wireData_t
+ * structures, each of which contains a deque for input and a
+ * deque for output. You can preload data to be read and you can
+ * look at a log of data that has been written.
+ */
 
 #pragma once
 
 #include <inttypes.h>
+
+// https://github.com/Arduino-CI/arduino_ci/issues/165
```

Comment:
I'm curious whether the `max` and `min` definitions here should replace my current definitions in the library itself -- whether they would pass all the existing tests.  I'm thinking of using this to resolve #133 (i.e. spreading this fix more broadly).

Reply:
I would very much like to have the fix spread more broadly (we are having to include it in several libraries)! I got the code from #144 (which might be languishing because it is too large!?).

Reply:
I hadn't considered merging #144 because the CI was failing -- I assumed it was still a work in progress.  I'll follow up on that, because I very much support the direction being taken.

Reply:
And now that I look more carefully, I see the most recent commit comment is "Debug commit - do not merge"!